### ISSUE TITLE
Hi, I trained your model but result not very good

### ISSUE THREAD

**Issue by jinfagang** [vanhuyz/CycleGAN-TensorFlow#18]
I run apple - > orange but when transfer apple to orange the result is very pool, using the default image size, I also trained the original pytorch version, it generates just nice picture, is I am not training enough epoch?

**Comment by vanhuyz** [comment]
Thank you for your information.
I've just realized that my implementation of PatchGAN was incorrect and I'm working on it.

Regards,

**Comment by jinfagang** [comment]
Thanks, this work is very meaningfull, cause original can not even predict single image, If you worked out, update repo let me know, I'd like to follow this project.

**Comment by cpury** [comment]
Hey man, thanks for your awesome repository. Everything is set up awesomely, Tensorboard and all, and I'd love to get this to work.

Unfortunately, I can confirm @jinfagang observation that the resulting models are not good at all. I spent like 20 USD on an AWS EC2 GPU instance until I realized this. Maybe until you fix this issue, mention this in the readme so other people don't waste their time as well.

Thanks!

**Comment by vanhuyz** [comment]
I've removed the wrong part of PatchGAN and now the results should be better. But it takes time to confirm.

**Comment by jinfagang** [comment]
@
Would you add some test image result for current made?  Have some sample result could be better I think.

**Comment by vanhuyz** [comment]
Results on training set after 37 epochs.

* apple2orange:
<img width="776" alt="apple2orange" src="https://cloud.githubusercontent.com/assets/7971458/26384572/42de991a-4075-11e7-87f9-658cec3b4bb2.png">

* orange2apple:
<img width="748" alt="orange2apple" src="https://cloud.githubusercontent.com/assets/7971458/26384571/42db012e-4075-11e7-81d0-7953ae4a2749.png">

**Comment by vanhuyz** [comment]
I tried on horse2zebra dataset but the results are still blurry. Maybe it is more difficult to translate horse2zebra than orange2apple.
Did you train horse2zebra with the original pytorch version?

**Comment by jinfagang** [comment]
I have trained original summer2winter and apple2orange, both get in some level fair result, at least the resolution are more clear.

**Comment by vanhuyz** [comment]
I fixed another issue and retrained horse2zebra. The results are quite promising.
A screenshot after 87 epochs:
<img width="750" alt="screen shot 2017-05-26 at 2 48 54 pm" src="https://cloud.githubusercontent.com/assets/7971458/26482482/ad1c9422-4222-11e7-8f47-24cbe87a6169.png">

**Comment by jinfagang** [comment]
@vanhuyz Hi, I also managed predict using original torch version and reconstruct a production version, here is the repo https://github.com/jinfagang/pytorch_cycle_gan, I have some results, seems original still more clear, what did you see? Maybe we can discuss about the difference detail, do you have wechat ?

**Comment by TX2012LH** [comment]
What does the `another issue` mean? Could you please point it out specifically?

**Comment by vanhuyz** [comment]
@TX2012LH just 1 line but critical https://github.com/vanhuyz/CycleGAN-TensorFlow/commit/960d1b085459856213d653d0f4e5a1c42dc1f0cf. G and F of cycle loss were reversed :flushed:

**Comment by vanhuyz** [comment]
@jinfagang Sorry I don't have wechat. How about via email?

**Comment by jinfagang** [comment]
This is my email nicholasjela@gmail.com, you can send me some information
